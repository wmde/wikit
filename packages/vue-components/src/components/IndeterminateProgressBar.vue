<template>
	<div class="wb-ui-indeterminate-progress-bar__box">
		<div
			class="wb-ui-indeterminate-progress-bar__bar"
			aria-busy="true"
			aria-live="polite"
		/>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';

@Component
export default class IndeterminateProgressBar extends Vue {
}
</script>

<style lang="scss">
.wb-ui-indeterminate-progress-bar__box {
	position: absolute;
	height: 20px;
	width: 90%;
	left: 5%;
	top: 50%;
	transform: translate( 0%, -50% );
	border: 1px solid #a2a9b1;
	box-shadow: 0 2px 0 0 rgba( 0, 0, 0, 0.15 );
	box-sizing: border-box;
	border-radius: 2px;
	overflow: hidden;
	background-color: #fff;
}

.wb-ui-indeterminate-progress-bar__bar {
	animation: wb-ui-indeterminate-progress-bar__bar--slide 2s infinite linear;
	background-color: $wmf-color-primary;
	width: 40%;
	height: 100%;
	display: block;
}

@keyframes wb-ui-indeterminate-progress-bar__bar--slide {
	from {
		transform: translate( -100% );
	}

	to {
		transform: translate( 350% );
	}
}
</style>
