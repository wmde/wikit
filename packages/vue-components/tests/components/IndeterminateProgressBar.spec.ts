import IndeterminateProgressBar from '@/components/IndeterminateProgressBar.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'IndeterminateProgressBar', () => {
	it( 'renders correctly', () => {
		const wrapper = shallowMount( IndeterminateProgressBar );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'mounted correctly', () => {
		const wrapper = shallowMount( IndeterminateProgressBar );
		expect( wrapper.isVueInstance() ).toBe( true );
	} );
} );
