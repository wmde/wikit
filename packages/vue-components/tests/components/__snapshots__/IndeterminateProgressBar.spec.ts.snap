// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndeterminateProgressBar renders correctly 1`] = `
<div
  class="wb-ui-indeterminate-progress-bar__box"
>
  <div
    aria-busy="true"
    aria-live="polite"
    class="wb-ui-indeterminate-progress-bar__bar"
  />
</div>
`;
